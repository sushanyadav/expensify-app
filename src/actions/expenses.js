import { v4 as uuidv4 } from 'uuid';

//ADD_EXPENSE

export const addExpense = (
    { description = '',
        note = '',
        amount = 0,
        createdAt = 0 } = {} //if no argument is provided its default to  empty object{}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuidv4(),
        description,
        note,
        amount,
        createdAt
    }
})

//REMOVE_EXPENSE

export const removeExpense = ({ id } = {}) => ({

    type: 'REMOVE_EXPENSE',
    id

})

//EDIT_EXPENSE

export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates //  amount: 1 
})

