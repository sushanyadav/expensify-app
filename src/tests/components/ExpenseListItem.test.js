import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import ExpenseListItem from '../../components/ExpenseListItem'

const expense = expenses[1]


test('should render Expense List item with given expense', () => {
    const wrapper = shallow(<ExpenseListItem id={expense.id} description={expense.description} amount={expense.amount} createdAt={expense.createdAt} />)
    expect(wrapper).toMatchSnapshot()
})