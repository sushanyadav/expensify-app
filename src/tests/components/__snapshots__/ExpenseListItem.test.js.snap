// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense List item with given expense 1`] = `
<div>
  <[object Object]
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </[object Object]>
  <p>
    109500
     - 
    -345600000
  </p>
</div>
`;
